openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -201,3 +206,29 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 	}
 	return route.Status.Ingress[0].Host
 }
+
+func (e *extension) cleanupDomainMapping(ctx context.Context, ks *operatorv1beta1.KnativeServing) error {
+	client := e.kubeclient
+
+	for _, dep := range []string{"domain-mapping", "domainmapping-webhook"} {
+		if err := client.AppsV1().Deployments(ks.GetNamespace()).Delete(ctx, dep, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete deployment %s: %w", dep, err)
+		}
+	}
+
+	// Delete the rest of the domain mapping resources
+	if err := client.CoreV1().Services(ks.GetNamespace()).Delete(ctx, "domainmapping-webhook", metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
We need to delete two more created by monitoring:

```
domain-mapping-sm-service          ClusterIP   172.30.94.81     <none>        8444/TCP                                  10m
domainmapping-webhook-sm-service   ClusterIP   172.30.12.108    <none>        8444/TCP                                  10m
```

Reply:
These services are owners of the following secrets:
```
domain-mapping-sm-service-tls          kubernetes.io/tls                     2      13m
domainmapping-webhook-sm-service-tls   kubernetes.io/tls                     2      13m
```
which they will be deleted once the services are deleted.

Reply:
Also there are a few leases that probably it would be good to delete:
```
domainmapping-webhook.defaultingwebhook.00-of-01                                                                                                                 14m
domainmapping-webhook.validationwebhook.00-of-01                                                                                                                 14m
domainmapping-webhook.webhookcertificates.00-of-01                                                                                                               14m
domainmapping.knative.dev.serving.pkg.reconciler.domainmapping.reconciler.00-of-01                                                                               14m
```

Reply:
Tested with 1.31 -> 1.32 LGTM minus the updates above.

Reply:
Thanks!